### ISSUE TITLE
Suggestion: Color presets for friends, CC, Chat-Channel, GIM, and unaffiliated players

### ISSUE THREAD

**Issue by Acronyte** [BagleBreath/beaglebreath-chat-name-colors#1]
Hi,

This plugin is very nice for differentiating names. To expand the functionality further, I was wondering how feasible it would be to add preset colors for unspecified users in friends list, cc, chat channel, etc. For example, if I haven't manually set a chat color for a clan member, then their name would show up as the preset color that I picked for clan members. This would be overridden though if you manually picked a color for a user. Perhaps the status of these users could also be checked periodically, such as if someone leaves the clan or chat-channel, then their color will be updated to the default color for normal players (not on friends list/cc/chat channel).

I would think the hierarchy for what color shows would be something like:
Manually Chosen Color > Friends List > Group Ironman > Clan Chat > Chat Channel > Unaffiliated players you don't know

Let me know what you think. Thanks

**Comment by BagleBreath** [comment]
Thank you for the thoughtful suggestion! This is a great idea and definitely think it is supportable

**Comment by McVay** [comment]
Are you open to PRs? I'd love to work on adding this functionality since I was about to raise the same issue :).

**Comment by BagleBreath** [comment]
> Are you open to PRs? I'd love to work on adding this functionality since I was about to raise the same issue :).

Absolutely! Would love to review any incoming PRs. Haven't found the time to do it myself just yet
